runtime/io.c

```diff
@@ -477,7 +484,9 @@ intnat caml_input_scan_line(struct channel *channel)
       /* Fill the buffer as much as possible */
       n = caml_read_fd(channel->fd, channel->flags,
                        channel->max, channel->end - channel->max);
-      if (n == Io_interrupted) goto again;
+      if (n == -1) {
+        if (errno == EINTR) goto again; else caml_sys_error(NO_ARG);
```

Comment:
In the `trunk` implementation (where the exception is raised by `caml_read_fd`), we use `caml_sys_io_error(NO_ARG)` which handles `EAGAIN` and `EWOULDBLOCK` specially, but here we use `caml_sys_error` which does not do that. What am I missing?

Reply:
My mistake.  Fixed in e0b42a67b9.